driver_darwin.go

```diff
@@ -181,7 +187,7 @@ func oto_render(inUserData unsafe.Pointer, inAQ C.AudioQueueRef, inBuffer C.Audi
 	defer t.Stop()
 	ch := t.C
 
-	for len(buf) < queueBufferSize {
+	for len(buf) < queueBufferSize && d.ctx.Err() == nil {
```

Comment:
When is the error set?

Reply:
when context cancel called, ctx.err will non zero. A equivalent check to force the loop stops on cancel called

Reply:
Ah, ok thanks